examples/02_miscellaneous/plot_1_pearson_correlation.py

```diff
@@ -97,32 +97,51 @@
 # We first define a function to compute the Pearson correlation
 # between two tensors. Such function is not available in PyTorch,
 # but it is easy to implement.
-def pearson_corr(x, y):
-    """Compute the Pearson correlation between two tensors."""
+def pearson_corr(x, y, plan):
+    """
+    Compute the Pearson correlation between transformed
+    source features and target features.
+    """
     if not torch.is_tensor(x) or not torch.is_tensor(y):
-        x = torch.tensor(x)
-        y = torch.tensor(y)
+        x = torch.tensor(x).to(torch.float64)
+        y = torch.tensor(y).to(torch.float64)
 
-    vx = x - torch.mean(x)
+    # Compute the transformed features
+    x_transformed = (
+        (plan @ x.T / plan.sum(dim=1).reshape(-1, 1)).T.detach().cpu()
+    )
+
+    vx = x_transformed - torch.mean(x_transformed)
     vy = y - torch.mean(y)
 
     corr = (
         torch.sum(vx * vy)
-        / (torch.sqrt(torch.sum(vx**2)) * torch.sqrt(torch.sum(vy**2)))
+        / (torch.sqrt(torch.sum(vx**2) * torch.sum(vy**2)))
     ).numpy()
 
     return corr
 
 
 # %%
 # We then define a callback function that computes the
-# Pearson correlation between the transformed features and
-# the target features at each BCD iteration.
+# Pearson correlation between transformed features and
+# target features at each BCD iteration.
+
+
+# Initialize the transport plan with ones and normalize it
+init_plan = torch.ones(
+    (source_features_normalized.shape[1], source_features_normalized.shape[1])
+).to(float)
+init_plan_normalized = init_plan / init_plan.sum()
```

Comment:
Great!
Now, I think you should use this plan when fitting your mapping (simply add `init_plan=init_plan` in `mapping.fit()`.
You and I know that what fugw does under the hood is to instantiate a plan equal to the one you just created, but this might change in the future.